src/lib/operators/abstract_operator.cpp

```diff
@@ -1,18 +1,20 @@
 #include "abstract_operator.hpp"
+#include "utils/assert.hpp"
 
 namespace opossum {
 
 AbstractOperator::AbstractOperator(const std::shared_ptr<const AbstractOperator> left,
                                    const std::shared_ptr<const AbstractOperator> right)
-    : _left_input(left), _right_input(right) {}
+    : _left_input(left), _right_input(right), _was_executed{false} {}
 
 void AbstractOperator::execute() {
+  Assert(!_was_executed, "Operators shall not be executed twice.");
   _output = _on_execute();
+  _was_executed = true;
 }
 
 std::shared_ptr<const Table> AbstractOperator::get_output() const {
-  // TODO(student): You should place some meaningful checks here
-
+  DebugAssert(!_was_executed || (_was_executed && _output), "Execute failed and returned nullptr.");
```

Comment:
This could return `nullptr` if we have not yet executed this operator.
```suggestion
  DebugAssert(!_was_executed ||  _output, "Execute failed and returned nullptr.");
```

Reply:
We do, as `_output` is initially null